airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -257,9 +268,7 @@ private void reportFailure(final ConnectionUpdaterInput connectionUpdaterInput,
       connectionUpdaterInput.setAttemptNumber(attemptNumber + 1);
       connectionUpdaterInput.setFromFailure(true);
     } else {
-      runMandatoryActivity(jobCreationAndStatusUpdateActivity::jobFailure, new JobFailureInput(
-          connectionUpdaterInput.getJobId(),
-          "Job failed after too many retries for connection " + connectionId));
+      runMandatoryActivity(jobCreationAndStatusUpdateActivity::jobFailure, new JobFailureInput(connectionUpdaterInput.getJobId(), failureReason));
```

Comment:
Based on how this is call (the ternary operator that returns null for the failure reason if it is not a config error), we will lose the failure reason.